Asgard/Communications/Interfaces/IEventManager.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using Asgard.Data.Interfaces;
+
+namespace Asgard.Communications
+{
+    public delegate void CbusEventCallback(ICbusAccessoryEvent cbusAccessoryEvent, ICbusEventManager cbusEventManager);
```

Comment:
As per similar thoughts on the `ResponseManager` callbacks - wondering whether the event manager needs to be in this callback - whoever has registered for a callback likely has a handy reference to it already anyway

Reply:
The ICbusEventManager being a parameter to the callback is so the concrete CbusEventManager can inject itself into the callback. That then allows the callback access to the state of the events that the CbusEventManager has observed.

Reply:
The code where the callback is defined isn't the place to provide access to the CbusEventManager. To do that the callback would have to capture the CbusEventManager, which then means that it has the data in at the time of capture, whereas it needs to use the current state at the time of the callback being run.

Reply:
It’d only be capturing the reference to the event manager - it can still query it for the current state, same as if it’s passed in the callback.  Can also capture and query the state of any other objects that would be necessary to make decisions on